node-src/lib/getFileHashes.ts

```diff
@@ -0,0 +1,71 @@
+import { close, open, read } from 'fs';
+import pLimit from 'p-limit';
+import { join } from 'path';
+import xxHashWasm, { XXHash, XXHashAPI } from 'xxhash-wasm';
+
+const hashFile = (buffer: Buffer, path: string, xxhash: XXHashAPI): Promise<string> => {
+  const BUFFER_SIZE = buffer.length;
+
+  return new Promise((resolve, reject) => {
+    const done = (fd: number, getResult: () => bigint) => {
```

Comment:
What does `fd` stand for?

Reply:
File descriptor. Pretty common in the Node.js world.

Reply:
Luckily we also have types to clarify things if needed. The `fd` variable is _so_ common that I don't think spelling it out will actually improve readability. The [official docs](https://nodejs.org/en/learn/manipulating-files/working-with-file-descriptors-in-nodejs) use `fd` everywhere and I've never seen anyone _not_ use `fd`.